### PR TITLE
Remove Survivor Revolvers

### PR DESCRIPTION
**Pull request by Holli-Git**
#### Summary
Bugfixes "Remove Survivor Revolvers"

#### Purpose of change

Survivor revolvers didn't make sense, as the cartridges lack a rim or moonclip, and a survivor wouldn't really be able to machine the specific dimensions of the cartridge into a cylinder to have it hold.

#### Describe the solution

Removes it.

#### Describe alternatives you've considered

Adding a 45 colt revolver, but then it becomes a question of would a survivor even be able to make a revolver, and so I settled on making a single shot 45 pistol in a future PR.

#### Testing

Tested locally, works fine.

### CODE REVIEW

**Comment by TheShadowFerret** [comment]
>Adding a 45 colt revolver, but then it becomes a question of would a survivor even be able to make a revolver, and so I settled on making a single shot 45 pistol in a future PR.

Nobody has worked on the necessary tools to do so(see #66060), so the answer is "eventually, but not now so no revolver."

**Comment by Holli-Git** [comment]
> > Adding a 45 colt revolver, but then it becomes a question of would a survivor even be able to make a revolver, and so I settled on making a single shot 45 pistol in a future PR.
> 
> Nobody has worked on the necessary tools to do so(see #66060), so the answer is "eventually, but not now so no revolver."

Yeah that's why I was going with a single shot break action pistol thing instead

**Comment by anoobindisguise** [comment]
I'm a big fan of the idea of the survivor-made revolvers. If it's too much of a stretch to use regular ammo (and regular revolvers are common enough that you wouldn't bother anyways) why not make em like, paper cartridge guns? Those are scarcer, there aren't many M1860s lying around but they serve a valuable niche with producible ammo. The six shooter and hand cannon could pretty easily be converted to .36 and .44 paper cartridge respectively. Granted we're missing some tools we need to make them properly but I'd personally be sad to see these go with no replacement.

**Comment by Holli-Git** [comment]
> I'm a big fan of the idea of the survivor-made revolvers. If it's too much of a stretch to use regular ammo (and regular revolvers are common enough that you wouldn't bother anyways) why not make em like, paper cartridge guns? Those are scarcer, there aren't many M1860s lying around but they serve a valuable niche with producible ammo. The six shooter and hand cannon could pretty easily be converted to .36 and .44 paper cartridge respectively. Granted we're missing some tools we need to make them properly but I'd personally be sad to see these go with no replacement.

I dislike the notion that you need to replace something that's obsoleted, it puts extra pressure on people obsoleting things. If something doesn't make sense, you shouldn't go "But you should add a replacement!", adding a replacement can be saved for another PR. I don't want to go through the effort of researching how revolvers are made in order to make the survivor revolvers make sense, and then still not used at all because they're just not useful.

Also, do people actually use these? At all? I'm a pretty big user of niche guns and I have never gone out of my way to craft or find one of these to use them, they're just not worth it.

File: data/json/obsoletion_and_migration_0.I/migration_items.json [hunk 1]
```diff
@@ -1221,5 +1221,15 @@
...
+  },
+  {
+    "id": "surv_six_shooter",
+    "type": "MIGRATION",
+    "replace": "sw_619"
+  },
+  {
+    "id": "surv_hand_cannon",
+    "type": "MIGRATION",
+    "replace": "sw_619"
```
Comment (tenmillimaster) [inline data/json/obsoletion_and_migration_0.I/migration_items.json:1233]: Wouldn't it be a cleaner migration to move to stuff in the same caliber?

Comment (Holli-Git) [reply data/json/obsoletion_and_migration_0.I/migration_items.json:1233]: > Wouldn't it be a cleaner migration to move to stuff in the same caliber?

What gun would be better? I don't imagine migrating to a 1911 would be any better, as it has no internal magazine

**Comment by Maleclypse** [comment]
> > I'm a big fan of the idea of the survivor-made revolvers. If it's too much of a stretch to use regular ammo (and regular revolvers are common enough that you wouldn't bother anyways) why not make em like, paper cartridge guns? Those are scarcer, there aren't many M1860s lying around but they serve a valuable niche with producible ammo. The six shooter and hand cannon could pretty easily be converted to .36 and .44 paper cartridge respectively. Granted we're missing some tools we need to make them properly but I'd personally be sad to see these go with no replacement.
> 
> I dislike the notion that you need to replace something that's obsoleted, it puts extra pressure on people obsoleting things. If something doesn't make sense, you shouldn't go "But you should add a replacement!", adding a replacement can be saved for another PR. I don't want to go through the effort of researching how revolvers are made in order to make the survivor revolvers make sense, and then still not used at all because they're just not useful.
> 


I do have to agree that it is unfortunately on the people who want a realistic way to make a thing to do the work of getting it back in game.  On occasion for something requiring massive c++ work the dev team goes the opposite direction, see UPS and repair kits.  But as this and everything required to make survivor revolvers possible is doable in json it wouldn’t be waived here.

**Pull request merged**